src/megaclient.cpp

```diff
@@ -1915,18 +1913,69 @@ void MegaClient::exec()
         }
 
         // handle API server-client requests
-        if (!jsonsc.pos && pendingsc && !loggingout)
+        if (pendingscUserAlerts)
+        {
+            switch (pendingscUserAlerts->status)
+            {
+            case REQ_SUCCESS:
+                if (*pendingscUserAlerts->in.c_str() == '{')
+                {
+                    JSON json;
+                    json.begin(pendingscUserAlerts->in.c_str());
+                    json.enterobject();
+                    if (useralerts.procsc_useralert(json))
+                    {
+                        // NULL vector: "notify all elements"
+                        app->useralerts_updated(NULL, int(useralerts.alerts.size()));
+                    }
+                    pendingscUserAlerts.reset();
+                    break;
+                }
+
+                // fall through
+            case REQ_FAILURE:
+                if (pendingscUserAlerts->httpstatus != 200)
+                {
+                    (pendingscUserAlerts->httpstatus == 500 ? fnstats.e500Count : fnstats.eOthersCount)++;
+                }
+                else
+                {
+                    error e = (error)atoi(pendingscUserAlerts->in.c_str());
+                    if (e == API_EAGAIN || e == API_ERATELIMIT)
+                    {
+                        fnstats.eAgainCount++;
+                        btsc.backoff();
+                        pendingscUserAlerts.reset();
```

Comment:
If we reset the `pnedingscUserAlerts`, it means that upon reception of these (retriable) errors from API during retrieval of UAs we will not retry at all, correct? Is it the expected behavior? In case the API is busy and returns a temporal -3, useralerts will not be fetched.

Reply:
The reset there is just to get back to the state before we sent this request - after the btsc timer has finished waiting and is armed again, the code to send the next wsc/sc request will be activated, and since useralerts.begincatchup is still true, it will send the UserAlerts request again.   If the error is not -3 (or ratelimit) then we abandon the useralerts request since it's not essential, by also setting useralerts.begincatchup false (and useralerts.catchupdone true).  After adjusting for the above comments, I've stepped through again to verify (and added some log warnings so we can see it in the logs if needed).  thanks!

Reply:
Oh, I see. My bad, I didn't think about the `begincatchup`. It makes perfect sense now :)